### PR TITLE
Resource property for enabling/disabling the ArcGIS Server service directory

### PR DESCRIPTION
**Pull request by PleaseStopAsking**
Created this since I did not see this specific task in any existing resources. 

Resource Usage:
```
Import-Module -Name ArcGIS
Import-DscResource -Name ArcGIS_Server_ServiceDir

ArcGIS_Server_ServiceDir disableServiceDir {
	Enabled = $false
	SiteName = 'arcgis'
	SiteAdministrator = $psaCredential
}

### CODE REVIEW

**Comment by nshampur** [comment]
Lets move this resource functionality into the ArcGIS_Server resource as an additional boolean property.  Also dbl. check the file encoding. github detects as binary.  This can be fixed using .gitattributes

**Comment by PleaseStopAsking** [comment]
I will make this change. Any issue with the property name ```ServicesDirectoryEnabled```?

**Comment by nshampur** [comment]
the name is fine

File: Modules/ArcGIS/DSCResources/ArcGIS_Server/ArcGIS_Server.psm1 [hunk 1]
```diff
@@ -987,10 +1018,86 @@ function Get-SingleClusterModeOnServer
...
+        $SettingUrl,
+
+        [System.String]
+        $Token,
+
+        [System.String]
+        $Referer
+    )
+
+    $RequestUrl  = $ServerUrl.TrimEnd("/") + "/" + $SettingUrl.TrimStart("/")
```
Comment (nshampur) [inline Modules/ArcGIS/DSCResources/ArcGIS_Server/ArcGIS_Server.psm1]: Can you please replace with the Invoke-ArcGISRequest utility method. Aside from this everything else looks great.

File: Modules/ArcGIS/DSCResources/ArcGIS_Server/ArcGIS_Server.psm1 [hunk 2]
```diff
@@ -987,10 +1018,86 @@ function Get-SingleClusterModeOnServer
...
+    {
+        $props['servicesDirEnabled'] = $props['enabled']
+    }
+    $cmdBody = To-HttpBody $props
+    $headers = @{'Content-type'='application/x-www-form-urlencoded'
+                'Content-Length' = $cmdBody.Length
+                'Accept' = 'text/plain'
+                'Referer' = $Referer
+                }
+    $res = Invoke-WebRequest -Uri $RequestUrl -Body $cmdBody -Method POST -Headers $headers -UseDefaultCredentials -DisableKeepAlive -UseBasicParsing
```
Comment (nshampur) [inline Modules/ArcGIS/DSCResources/ArcGIS_Server/ArcGIS_Server.psm1]: same - replace with Invoke-ArcGISRequest

**Comment by nshampur** [comment]
@shailesh91  .. plz merge when convenient

**Comment by PleaseStopAsking** [comment]
@shailesh91 Fixed encoding as requested.

**Comment by PleaseStopAsking** [comment]
@shailesh91 Looks like this commit is missing at v2.0-dev release. Will it be added back at full release?

**Comment by shailesh91** [comment]
@PleaseStopAsking this commit was missed in the release I will create a PR soon to add this back.

**Pull request merged**